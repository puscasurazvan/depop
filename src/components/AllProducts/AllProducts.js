import React, { useEffect } from 'react'

import PropTypes from 'prop-types'

import Product from '../Product'

const AllProducts = ({ products, setCount }) => {
  const allProducts = products.map(item => item)

  useEffect(() => {
    if (allProducts) {
      setCount(allProducts.length)
    }
  }, [setCount, allProducts])

  return products.map(({ id, title, brand, size, price, img, sold }) => {
    return (
      <Product
        sold={sold}
        key={id}
        productTitle={title}
        imageURL={img}
        productBrand={brand || 'No brand specified'}
        productSize={size}
        productPrice={price}
      />
    )
  })
}

AllProducts.propTypes = {
  products: PropTypes.array.isRequired,
  setCount: PropTypes.func,
}

export default AllProducts
