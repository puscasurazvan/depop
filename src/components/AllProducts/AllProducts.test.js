import React from 'react'
import { mount } from 'enzyme'
import AllProducts from './AllProducts'

const minProps = {
  products: [],
  setCount: jest.fn(),
}

describe('AllProducts', () => {
  it('should render with the right props', () => {
    const wrapper = mount(<AllProducts {...minProps} />)
    expect(wrapper.props().products).toStrictEqual([])
    expect(wrapper.find('AllProducts').length).toEqual(1)
  })

  it('should render without exploding', () => {
    const wrapper = mount(<AllProducts {...minProps} />)
    expect(wrapper.length).toEqual(1)
    expect(wrapper).toMatchSnapshot()
  })
})
