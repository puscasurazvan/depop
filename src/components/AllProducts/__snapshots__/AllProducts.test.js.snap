// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllProducts should render without exploding 1`] = `ReactWrapper {}`;
